From 264fa9ac12f567cc50043eaebc5b439d3dd92440 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Wed, 18 May 2022 18:16:45 +0200
Subject: [PATCH] setup-disk: use smaller ESP partition for UEFI

Grub's efi files are just a few MB, so we can reduce the size of the
ESP. We also enforce FAT32, which requires minimum 32MB. Apparently many
UEFI firmware implementations only support FAT32.

https://wiki.osdev.org/EFI_System_Partition#Format
---
 setup-disk.in | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/setup-disk.in b/setup-disk.in
index b14bbe9..200393d 100644
--- a/setup-disk.in
+++ b/setup-disk.in
@@ -905,8 +905,9 @@ setup_boot_dev() {
 		bootdev=/dev/md0
 	fi
 	case "$BOOTFS" in
-	btrfs) mkfs_args="";;
-	ext4) mkfs_args="$mkfs_args -O ^64bit";; # pv-grub does not support 64bit
+		vfat) mkfs_args="-F 32";; # UEFI firmware often only support FAT32
+		btrfs) mkfs_args="";;
+		ext4) mkfs_args="$mkfs_args -O ^64bit";; # pv-grub does not support 64bit
 	esac
 	mkfs.$BOOTFS $MKFS_OPTS_BOOT $mkfs_args $bootdev
 	BOOT_DEV="$bootdev"
@@ -1462,7 +1463,7 @@ fi
 if is_efi || [ -n "$USE_EFI" ]; then
 	USE_EFI=1
 	DISKLABEL=gpt
-	BOOT_SIZE=${BOOT_SIZE:-512}
+	BOOT_SIZE=${BOOT_SIZE:-36} # UEFI needs fat32, which needs minimum 32MB
 	BOOTFS=vfat
 	: ${BOOTLOADER:=grub}
 fi
-- 
2.36.1

